src/main/java/org/truffleruby/core/hash/PreInitializationManager.java

```diff
@@ -26,13 +27,26 @@
 
     private final RubyContext context;
 
+    private final List<ReHashable> reHashables = new ArrayList<>();
+
     private TrackingHashFactory trackingHashFactory;
     private final Set<DynamicObject> hashesCreatedDuringPreInit = Collections.newSetFromMap(new WeakHashMap<>());
 
     public PreInitializationManager(RubyContext context) {
         this.context = context;
     }
 
+    public void addReHashable(ReHashable reHashable) {
+        // This might get called multiple times for the same ReHashable,
+        // so only add it if it is not already in the List.
```

Comment:
Why not use a `Set` then? As far as I can tell, nothing here cares about ordering.

Reply:
I want it deterministic for debug-ability, and it seems better to always rehash the RopeCache first before the SymbolTable as they sort of depend on each other.